core/src/test/scala/unit/kafka/log/LogSegmentsTest.scala

```diff
@@ -238,5 +248,39 @@ class LogSegmentsTest {
     assertEquals(Int.MaxValue, LogSegments.sizeInBytes(asList(logSegment)))
     assertEquals(largeSize, LogSegments.sizeInBytes(asList(logSegment, logSegment)))
     assertTrue(UnifiedLog.sizeInBytes(asList(logSegment, logSegment)) > Int.MaxValue)
+
+    val logSegments: LogSegments = new LogSegments(topicPartition)
+    logSegments.add(logSegment)
+    assertEquals(Int.MaxValue, logSegments.sizeInBytes())
+  }
+
+  @Test
+  def testUpdateDir(): Unit = {
+    val seg1 = createSegment(1)
+    val segments = new LogSegments(topicPartition)
+    segments.add(seg1)
+
+    val newDir: File = TestUtils.tempDir()
+    segments.updateParentDir(newDir)
+    assertEquals(newDir, seg1.log().file().getParentFile)
+    assertEquals(newDir, seg1.timeIndexFile().getParentFile)
+    assertEquals(newDir, seg1.offsetIndexFile().getParentFile)
+    assertEquals(newDir, seg1.txnIndex().file().getParentFile)
+
+    Utils.delete(newDir)
+  }
+
+  @Test
+  def testGetFirstBatchTimestampForSegments(): Unit = {
+    val segments: java.util.List[LogSegment] = new util.ArrayList[LogSegment]()
+    val seg1 = createSegment(1)
+    val seg2 = createSegment(2)
+    segments.add(seg1)
+    segments.add(seg2)
+    assertEquals(Seq(Long.MaxValue, Long.MaxValue), LogSegments.getFirstBatchTimestampForSegments(segments).asScala.toSeq)
```

Comment:
It seems `getFirstBatchTimestampForSegments` is used by `UnifiedLog` only, so we should move this helper to `UnifiedLog`

https://github.com/apache/kafka/blob/bac8df56ffdf8a64ecfb78ec0779bcbc8e9f7c10/core/src/main/scala/kafka/log/UnifiedLog.scala#L1921